torch/fx/experimental/GraphManipulation.py

```diff
@@ -61,18 +69,114 @@ def get_size_of_node(fx_module: GraphModule, node: Node) -> size_bytes:
+    shape_str = shape_str[:-2] + "]"
+    return shape_str
+
+
+def serialize_tensor_quantization(tensor: torch.Tensor) -> Dict:
+    scheme = {}
+    if tensor.is_quantized:
+        scheme["q_scheme"] = torch.qscheme
+        ##TODO finish quantized serialization
+    return scheme
+
+
+def serialize_weight(tensor: torch.Tensor) -> Dict:
+    weight = {}
+    weight["type"] = str(tensor.dtype)
+    weight["is_quantized"] = tensor.is_quantized
+    if tensor.is_quantized:
+        weight["quantized_type"] = serialize_tensor_quantization(tensor)
+    weight["shape"] = serialize_shape(tensor.shape)
+    return weight
+
+
+def serialize_module(fx_module: GraphModule, weights, name_prefix="") -> Dict:
+    serialized_dict = {}
+    serialized_dict["modules"] = {}
+    serialized_dict["weights"] = {}
+    serialized_dict["nodes"] = []
+    parameters = fx_module.named_parameters()
+    for name, p in parameters:
+        if isinstance(p, torch.Tensor):
+            weight = serialize_weight(p)
+            prefix = ""
+            if name_prefix:
+                prefix = name_prefix + "."
+            serialized_dict["weights"][prefix + name] = weight
+            weights[prefix + name] = p
+    for node in fx_module.graph.nodes:
+        node_rep = {}
+        # Get shape/type info, currently not needed for call_module.
+        if node.op != "call_module":
+            shape = getattr(node, "shape", None)
+            if shape:
+                node_rep["shape"] = serialize_shape(shape)
+            else:
+                raise RuntimeError("Node has no shape attr")
+            dtype = getattr(node, "dtype", None)
+            if dtype:
+                node_rep["type"] = str(dtype)
+            else:
+                raise RuntimeError("Node has no dtype attr")
+
+        # Recurse down into any submodules we are calling.
+        if node.op == "call_module":
+            if isinstance(getattr(fx_module, node.target), torch.fx.GraphModule):
+                output = serialize_module(
+                    getattr(fx_module, node.target), weights, node.target
+                )
+                serialized_dict["modules"][node.target] = output
+
+        if node.op == "call_function":
+            node_rep["target"] = str(node.target.__name__)
+        else:
+            node_rep["target"] = str(node.target)
+
+        # Make sure we capture all constants.
+        if node.op == "get_attr":
+            target = getattr(fx_module, node.target)
+            if isinstance(target, torch.Tensor):
+                weight = serialize_weight(target)
+                prefix = ""
+                if name_prefix:
+                    prefix = name_prefix + "."
+                serialized_dict["weights"][prefix + node.target] = weight
+                weights[prefix + node.target] = target
+
+        node_rep["op_code"] = node.op
+        node_rep["name"] = node.name
+        node_rep["args"] = str(node.args)
+        node_rep["kwargs"] = str(node.kwargs)
```

Comment:
Unless you're planning on writing Python syntax parsing on the Glow runtime side, I think you're going to want to explode `args` and `kwargs` into corresponding JSON types. This should work directly since `tuple`, `list`, and `dict` are built-in supported JSON-serializable types. The only difficulty is dealing with nodes, which you can do with `map_arg`:

```
from torch.fx.node import map_arg

<...>

        node_rep["args"] = map_arg(node.args, str)
        node_rep["kwargs"] = map_arg(node.kwargs, str)
```

Reply:
On further thought, I think there might also be ambiguity in the JSON form between `str` arguments and `Node` arguments serialized as `str`. Is there a way we can define a special type in JSON to signify a node reference? Likely via a string contained within this object

Reply:
what do you think about prepending NODE_ to the string for nodes, e.g. NODE_submodule_0?

Reply:
I'd prefer to avoid any possibility of name collisions if possible. What about a convention for Nodes like a dict with two keys that looks like this:

```
{
  "is_node": true,
  "name": "my_node_name"
}
```

Reply:
Avoiding collisions makes sense, I was talking with @jfix71  about this a bit, do we need to do any differentiating at all. Essentially can't we always know on the deserializing side whether it's a node or string by context? 
Is there an example where that's not the case? An node that has a arg that could be string or node?

Reply:
Yeah, I'm guessing it likely won't make a difference, but definitely think it's good to be explicit when we can be. Out of curiosity though, are there any cases where an op takes an arg that could be either a node or a string, and the callee checks what the type of the arg object passed in is/behaves differently accordingly?